### DISCUSSION TITLE
Multitenant

### DISCUSSION THREAD

**Discussion by develocode777** [serverpod/serverpod#2003] [Q&A] [answered]
Hello! I really like this framework very much but I need multitenant support. After working with it for a few days I found that I can change the ports and the database name in config files and then I can use it in a multitenant way sort of. Is this a viable solution? Are there any drawbacks?

**Comment by SandPod** [comment]
Hello,

I might not fully understand the question, but it is intended that you can specify what database to connect to from the configuration files.

Could you share a bit more about your architecture?

**Reply by develocode777** [reply]
Hello,

We need a database for each tenant, the tables will all be the same but they need to be separated. The solution of different config files for each tenant seems a bit hacky, I don't know if the framework was intended to be used that way and I was asking if there are other risks with this solution.

**Reply by SandPod** [reply, marked as answer]
It sounds like you will just deploy multiple versions of the same ~~app~~ server?

That should be totally fine. I would just make sure that each project has their own secrets so that you don't share any database passwords or authentication keys between the instances.

**Reply by develocode777** [reply]
Yes, it will be the same app that will connect to a different server port for each tenant
